src/action/config/ActionConfiguration.cpp

```diff
@@ -225,27 +220,29 @@ void ActionConfiguration::createAction()
 
   // Determine data and mesh
   std::vector<int> sourceDataIDs;
-  int              targetDataID = -1;
-  mesh::PtrMesh    mesh;
-  for (const mesh::PtrMesh &aMesh : _meshConfig->meshes()) {
-    if (aMesh->getName() == _configuredAction.mesh) {
-      mesh = aMesh;
-      for (const mesh::PtrData &data : mesh->data()) {
-        if (std::find(_configuredAction.sourceDataVector.begin(), _configuredAction.sourceDataVector.end(), data->getName()) != _configuredAction.sourceDataVector.end()) {
-          sourceDataIDs.push_back(data->getID());
-        }
-        if (data->getName() == _configuredAction.targetData) {
-          targetDataID = data->getID();
-        }
-      }
+  int              targetDataID;
+  PRECICE_CHECK(_meshConfig->hasMeshName(_configuredAction.mesh),
+                "Data action uses mesh \"{}\" which is not configured. Please ensure that the correct mesh name is given in <action:python mesh=\"...\">", _configuredAction.mesh);
+  mesh::PtrMesh mesh = _meshConfig->getMesh(_configuredAction.mesh);
+
+  if (mesh->hasDataName(_configuredAction.targetData)) {
+    targetDataID = mesh->data(_configuredAction.targetData)->getID();
+    PRECICE_ASSERT(targetDataID != -1);
+    PRECICE_ASSERT(!_configuredAction.targetData.empty())
+  } else {
+    PRECICE_CHECK(_configuredAction.targetData.empty(),
```

Comment:
If `not mesh->hasDataName(_configuredAction.targetData)`, then `_configuredAction.targetData` has to be an empty string.